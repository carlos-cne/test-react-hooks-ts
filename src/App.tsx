import React, { useEffect, useState } from "react";

interface Todo {
  id?: number;
  name: string;
}

const App: React.FC = () => {
  const [todo, setTodo] = useState<Todo>({ name: "" });
  const [todos, setTodos] = useState<Todo[]>([]);

  async function addTodo(e: any) {
    console.log(todos);
    e.preventDefault();
    await setTodos([...todos, { ...todo, id: Math.random() }]);
    localStorage.setItem("@Todos", JSON.stringify(todos));
    setTodo({ name: "" });
    console.log(todos);
  }

  function removeTodo(idx: number | undefined) {
    idx &&
      setTodos(todos.filter(item => item.id !== idx)) &&
      localStorage.setItem("@Todos", JSON.stringify(todos));
  }

  useEffect(() => {
    const data: any = localStorage.getItem("@Todos");

    setTodos(JSON.parse(data));
  }, []);

  return (
    <div className="App">
      <h1>Teste usando o React Hooks com TypeScript</h1>
      <h4>Todos</h4>
      <form onSubmit={e => addTodo(e)}>
        <input
          id="inputTodo"
          type="text"
          placeholder="Digite o Todo"
          value={todo.name}
          onChange={e => setTodo({ name: e.target.value })}
        />
        <button name="buttonSubmitTodo" type="submit">
          Adiconar
        </button>
      </form>
      <div>
        <ul>
          {todos.map((item: Todo) => (
            <li key={item.id}>
              {item.name} -{" "}
              <button
                name="buttonRemoveTodo"
                onClick={() => removeTodo(item.id)}
              >
                Remover
              </button>
            </li>
          ))}
        </ul>
      </div>
    </div>
  );
};

export default App;
