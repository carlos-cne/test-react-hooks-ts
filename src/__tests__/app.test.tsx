import React from "react";
import { render, cleanup } from "react-testing-library";
import "jest-dom/extend-expect";

import App from "../App";

afterEach(cleanup);

describe("My first test", () => {
  it("Component App", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });
  it("Should have button submit", () => {
    const { findByTestId } = render(<App />);

    expect(findByTestId("inputTodo")).toBe(true);
  });
});
