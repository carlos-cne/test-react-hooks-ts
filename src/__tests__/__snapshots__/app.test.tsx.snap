// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My first test Component App 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Teste usando o React Hooks com TypeScript
    </h1>
    <h4>
      Todos
    </h4>
    <form>
      <input
        name="inputTodo"
        placeholder="Digite o Todo"
        type="text"
        value=""
      />
      <button
        name="buttonSubmitTodo"
        type="submit"
      >
        Adiconar
      </button>
    </form>
    <div>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;
